packages/inference/src/providers/providerHelper.ts

```diff
@@ -0,0 +1,137 @@
+import type { ChatCompletionOutput, TextGenerationOutput } from "@huggingface/tasks";
+import { InferenceOutputError } from "../lib/InferenceOutputError";
+import type { BodyParams, HeaderParams, UrlParams } from "../types";
+import { toArray } from "../utils/toArray";
+/**
+ * Base class for task-specific provider helpers
+ */
+export abstract class TaskProviderHelper {
+	constructor(
+		private provider: string,
+		private baseUrl: string,
+		private task?: string,
+		readonly clientSideRoutingOnly: boolean = false
+	) {}
+
+	/**
+	 * Return the response in the expected format.
+	 * Needs to be implemented in the subclasses.
+	 */
+	abstract getResponse(
```

Comment:
Currently used by some tasks but not others. Should we unify this to always use `getResponse` (might save some headaches in the future ^^)